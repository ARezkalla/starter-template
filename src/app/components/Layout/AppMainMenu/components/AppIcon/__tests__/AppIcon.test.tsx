import { renderWithProviders } from "@/app/test/renderWithProvider"
import { AppIcon } from "../"
import { it, describe } from "vitest"

describe("AppIcon", () => {
  it("should render", () => {
    const { container } = renderWithProviders(
      <AppIcon iconCode="code" sx={{}} />,
    )
    expect(container).toMatchSnapshot()
  })
  it("should render with question mark", () => {
    const { container } = renderWithProviders(<AppIcon iconCode="" sx={{}} />)
    expect(container).not.toMatchSnapshot()
  })
})
