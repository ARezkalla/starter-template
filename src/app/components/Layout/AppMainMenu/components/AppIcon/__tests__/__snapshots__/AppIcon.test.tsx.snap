// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppIcon > should render 1`] = `
<div>
  <div
    class="MuiBox-root css-5azvwx"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-1btd5gz-MuiIcon-root"
    >
      code
    </span>
  </div>
</div>
`;

exports[`AppIcon > should render with question mark 1`] = `
<div>
  <div
    class="MuiBox-root css-5azvwx"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
    >
      question_mark
    </span>
  </div>
</div>
`;
